plugins/nf-nomad/src/test/nextflow/nomad/NomadConfigSpec.groovy

```diff
@@ -181,4 +181,21 @@ class NomadConfigSpec extends Specification {
         config.jobOpts.affinitySpec.getValue() == '3'
         config.jobOpts.affinitySpec.getWeight() == 50
     }
+
+    void "should instantiate a constraint spec if specified"() {
+        when:
+        def config = new NomadConfig([
+                jobs: [constraint : {
+                    attribute '${meta.my_custom_value}'
+                    operator  ">"
+                    value     "3"
+                }]
```

Comment:
For now, a direct mapping is great since it works with mental model of Nomad users as per the job definitions https://developer.hashicorp.com/nomad/docs/job-specification/constraint 

Though for future, maybe this is something that maybe we can parse in a more user-friendly manner using a small DSL, which can have the added benefit that we can add more Nextflow level checks for the constraints being assigned correctly. What do you tihnk @jagedn , something worth tracking ?


Otherwise, in failing cases, Nomad layer would give an error code/message to the NF layer and we'd have to parse that.

Reply:
totally agree

as reminder, we need also to work in the `process` configuration as current `constraint` and `affinity` is for all tasks